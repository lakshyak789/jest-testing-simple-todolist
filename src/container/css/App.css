.app-class {
  text-align: center;
}
button {
  outline: none;
}
.wrapper .text-box {
  height: 30px;
  width: 240px;
  border: 1px solid grey;
  font-size: 18px;
}

.wrapper {
  padding: 20px 0;
}

.wrapper .add-todo {
  padding-left: 30px;
  background-color: rgb(0, 56, 58);
  color: white;
  padding: 10px 30px;
  margin-left: 10px;
}
.todo-list {
  background-color: rgb(224, 224, 224);
  display: inline-block;
  width: 360px;
}
ul li {
  list-style: none;
}
.list-items {
  font-size: 20px;
}
.list-items {
  font-size: 24px;
  padding-bottom: 20px;
}
.remove-item {
  float: right;
  margin-right: 30px;
  color: rgb(48, 4, 4);
  border: 1px solid rgb(209, 202, 202);
  padding: 8px 15px;
  background-color: rgb(172, 167, 167);
}
