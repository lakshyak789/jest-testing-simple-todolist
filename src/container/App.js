import React, { Component } from "react";
import "./css/App.css";
class App extends Component {
  state = {
    idcounter: 0,
    text: "",
    todos: []
  };
  addTodos() {
    if (this.state.text !== "") {
      const todoarr = this.state.todos;
      let id = this.state.todos.map(todos => todos.id);
      let maxids = id.length > 0 ? Math.max(...id) : 0;
      todoarr.push({ id: maxids + 1, text: this.state.text });
      this.setState({ todos: todoarr });
      this.setState({ text: "" });
    }
  }
  handleChange(e) {
    this.setState({ text: e.target.value });
  }
  removeitem(e) {
    const todos = this.state.todos.filter(todos => todos.id != e.target.id);
    this.setState({ todos });
    console.log(e.target.id);
  }
  render() {
    const renderlist = this.state.todos.map(todolist => {
      return (
        <li className="list-items" key={todolist.id}>
          {todolist.text}
          <button
            id={todolist.id}
            className="remove-item"
            onClick={this.removeitem.bind(this)}
          >
            Remove
          </button>
        </li>
      );
    });
    return (
      <div className="app-class">
        <h1>Add Todos to List</h1>
        <hr />
        <div className="wrapper">
          <input
            type="text"
            className="text-box"
            value={this.state.text}
            onChange={this.handleChange.bind(this)}
          />
          <button className="add-todo" onClick={this.addTodos.bind(this)}>
            Add Todo
          </button>
        </div>
        <h2>Todo Items:</h2>
        <div className="todo-list">
          <ul>{renderlist}</ul>
        </div>
      </div>
    );
  }
}
export default App;
