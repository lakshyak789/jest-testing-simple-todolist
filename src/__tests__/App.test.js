import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "../container/App";
configure({ adapter: new Adapter() });
const app = shallow(<App />);
describe("App component", () => {
  it("should render properly", () => {
    expect(app).toMatchSnapshot();
  });
  describe("on click of add button with no value", () => {
    const text = "";
    beforeEach(() => {
      app.find(".text-box").simulate("change", { target: { value: text } });
      app.find(".add-todo").simulate("click");
    });
    it("should not let change the value of state", () => {
      expect(app.state().todos).toEqual([]);
    });
  });
  describe("on click of add button with value", () => {
    const text = "google";
    const todos = [{ id: 1, text }];
    beforeEach(() => {
      app.find(".text-box").simulate("change", { target: { value: text } });
      app.find(".add-todo").simulate("click");
    });
    it("should add the item to the state", () => {
      expect(app.state().todos).toEqual(todos);
    });
  });
  describe("on click of remove button", () => {
    beforeEach(() => {
      app.find(".remove-item").simulate("click", { target: { id: 1 } });
    });
    it("should remove the item from the list", () => {
      expect(app.state().todos).toEqual([]);
    });
  });
});
