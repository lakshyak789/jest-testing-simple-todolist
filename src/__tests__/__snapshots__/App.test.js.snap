// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Add Todos to List
        </h1>,
        <hr />,
        <div
          className="wrapper"
        >
          <input
            className="text-box"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="add-todo"
            onClick={[Function]}
          >
            Add Todo
          </button>
        </div>,
        <h2>
          Todo Items:
        </h2>,
        <div
          className="todo-list"
        >
          <ul />
        </div>,
      ],
      "className": "app-class",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Todos to List",
        },
        "ref": null,
        "rendered": "Add Todos to List",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="text-box"
              onChange={[Function]}
              type="text"
              value=""
            />,
            <button
              className="add-todo"
              onClick={[Function]}
            >
              Add Todo
            </button>,
          ],
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "text-box",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Todo",
              "className": "add-todo",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add Todo",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Todo Items:",
        },
        "ref": null,
        "rendered": "Todo Items:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul />,
          "className": "todo-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Add Todos to List
          </h1>,
          <hr />,
          <div
            className="wrapper"
          >
            <input
              className="text-box"
              onChange={[Function]}
              type="text"
              value=""
            />
            <button
              className="add-todo"
              onClick={[Function]}
            >
              Add Todo
            </button>
          </div>,
          <h2>
            Todo Items:
          </h2>,
          <div
            className="todo-list"
          >
            <ul />
          </div>,
        ],
        "className": "app-class",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Todos to List",
          },
          "ref": null,
          "rendered": "Add Todos to List",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="text-box"
                onChange={[Function]}
                type="text"
                value=""
              />,
              <button
                className="add-todo"
                onClick={[Function]}
              >
                Add Todo
              </button>,
            ],
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "text-box",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Todo",
                "className": "add-todo",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add Todo",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Todo Items:",
          },
          "ref": null,
          "rendered": "Todo Items:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul />,
            "className": "todo-list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
